src/no-side-effects.spec.ts

```diff
@@ -85,6 +85,58 @@ createTester().run(ruleId, rule, {
       options: [{ excludedIdentifiers: ['assert', 'debug', 'log'] }],
       name: 'Valid case with no exports with assertions and schema validation',
     },
+    {
+      code: `
+           import debug from 'debug';
+           import Koa from 'koa';
+           import { v4 as uuid } from 'uuid';
+           import Router from '@koa/router';
+           import { StatusCodes } from 'http-status-codes';    
+           
+           const log = debug('report:event');
+           
+            assert(\`I'm a number, \${numberValue}\`);
+            assert.ok(statusCode === StatusCodes.OK, 'Status code is not OK');
+            Symbol.for('foo');
+            const object = {
+                  prop: 'foo'
+            };
+              
+            Object.freeze(object);
+            
+            try {
```

Comment:
i feel everything about this try/catch code should be an error with this rule?  i.e. usage of try is an error, and object.prop = 'bar' is also an error.